nav2_motion_primitives/src/back_up.cpp

```diff
@@ -108,4 +126,55 @@ nav2_tasks::TaskStatus BackUp::controlledBackup()
   return TaskStatus::FAILED;
 }
 
+bool BackUp::pathIsClear()
```

Comment:
I still don't fully understand this API - I agree that there's probably a need in most motions to check for collision, but this API doesnt give it a set of waypoints, or a time to project forward, or any important information to make that decision based on. This function in backup doesn't actually produce a `return true` and requests the pose that somewhere else in the call was probably determined to pass along

Reply:
I assume you are talking about the `WorldModel` API, not the `MotionPrimitive` base class. In the case of the API for checking for free space I tried to consider the simplest useful case, i.e. check within some rectangular region and return false if any cell is occupied, in this case it's up to the client to define the region. We can later extend the API with a more sofisticated approach like providing waypoints to the World Model (instead of the description of the rectangular region).

> This function in backup doesn't actually produce a return true and requests the pose that somewhere else in the call was probably determined to pass along

Could you clarify the second part of the question?